backend/src/test/java/de/neuefische/driver/backend/controllers/DeliveryIntegrationTest.java

```diff
@@ -1,27 +1,84 @@
 package de.neuefische.driver.backend.controllers;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.neuefische.driver.backend.models.Delivery;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.web.servlet.MockMvc;
 
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.fail;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
 
 @SpringBootTest
 @AutoConfigureMockMvc
 class DeliveryIntegrationTest {
     @Autowired
     MockMvc mockMvc;
 
+    @Autowired
+    ObjectMapper objectMapper;
+
     @Test
     void getDeliveries_shouldReturnEmptyList_whenRepoIsEmpty() throws Exception {
-        mockMvc.perform(get("/api/deliveries"))
+        mockMvc.perform(get("/api/deliveries")).andExpect(status().isOk()).andExpect(content().json("""
+                []
+                """));
+    }
+
+    @Test
+    void getDeliveryById_shouldThrowException_whenInvalidId() {
+        try {
+            mockMvc.perform(get("/api/deliveries/123"));
+            fail();
+        } catch (Exception e) {
+            if (!(e.getCause() instanceof NoSuchElementException)) {
+                fail();
+            }
+        }
+    }
+
+    @Test
+    @DirtiesContext
+    void addDelivery_shouldReturnAddedDelivery() throws Exception {
+        mockMvc.perform(post("/api/deliveries")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                                {"title":  "test"}
+                                """))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                        {"title":  "test"}
+                        """))
+                .andExpect(jsonPath("$.id").isNotEmpty());
+    }
```

Comment:
Kleiner Tipp für später, wenn die Tests bisschen umfangreicher werden: die json-strings für .content() und .andExpect(content().json()) kann man auch sehr gern in variablen auslagern um den code bisschen aufgeräumter zu bekommen.